File: DaS.ScriptLib.CS/Game/Data/Structures/EntityLocation.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using DaS.ScriptLib.Injection;

namespace DaS.ScriptLib.Game.Data.Structures
{

	public class EntityLocation
	{


		public readonly int Pointer;
		public EntityLocation(int ptr)
		{
			Pointer = ptr;
		}

		public void CopyFrom(EntityLocation other)
		{
			var oAngle = other.Angle;
			var oX = other.X;
			var oY = other.Y;
			var oZ = other.Z;

			Angle = oAngle;
			X = oX;
			Y = oY;
			Z = oZ;
		}

		public double Heading {
			get { return (Hook.RFloat(Pointer + 0x4) / Math.PI * 180) + 180; }
			set { Hook.WFloat(Pointer + 0x4, (float)(value * Math.PI / 180) - (float)Math.PI); }
		}

		public double Angle {
			get { return Hook.RFloat(Pointer + 0x4); }
			set { Hook.WFloat(Pointer + 0x4, (float)value); }
		}

		public float X {
			get { return Hook.RFloat(Pointer + 0x10); }
			set { Hook.WFloat(Pointer + 0x10, value); }
		}

		public float Y {
			get { return Hook.RFloat(Pointer + 0x14); }
			set { Hook.WFloat(Pointer + 0x14, value); }
		}

		public float Z {
			get { return Hook.RFloat(Pointer + 0x18); }
			set { Hook.WFloat(Pointer + 0x18, value); }
		}

	}

}

File: DaS.ScriptLib.CS/Game/Data/Structures/EntityLocation.vb
Imports DaS.ScriptLib.Injection

Namespace Game.Data.Structures

    Public Class EntityLocation

        Public ReadOnly Pointer As Integer

        Public Sub New(ptr As Integer)
            Pointer = ptr
        End Sub

        Public Sub CopyFrom(other As EntityLocation)
            Dim oAngle = other.Angle
            Dim oX = other.X
            Dim oY = other.Y
            Dim oZ = other.Z

            Angle = oAngle
            X = oX
            Y = oY
            Z = oZ
        End Sub

        Public Property Heading As Double
            Get
                Return (RFloat(Pointer + &H4) / Math.PI * 180) + 180
            End Get
            Set(value As Double)
                WFloat(Pointer + &H4, CType(value * Math.PI / 180, Single) - CType(Math.PI, Single))
            End Set
        End Property

        Public Property Angle As Double
            Get
                Return RFloat(Pointer + &H4)
            End Get
            Set(value As Double)
                WFloat(Pointer + &H4, CType(value, Single))
            End Set
        End Property

        Public Property X As Single
            Get
                Return RFloat(Pointer + &H10)
            End Get
            Set(value As Single)
                WFloat(Pointer + &H10, value)
            End Set
        End Property

        Public Property Y As Single
            Get
                Return RFloat(Pointer + &H14)
            End Get
            Set(value As Single)
                WFloat(Pointer + &H14, value)
            End Set
        End Property

        Public Property Z As Single
            Get
                Return RFloat(Pointer + &H18)
            End Get
            Set(value As Single)
                WFloat(Pointer + &H18, value)
            End Set
        End Property

    End Class

End Namespace
